packages/client-core/src/user/services/AuthService.ts

```diff
@@ -144,17 +145,17 @@ const waitForToken = async (win, clientUrl): Promise<string> => {
     )
     const getIframeResponse = function (e) {
       if (e.origin !== clientUrl) return
-      if (e?.data) {
+      if (e?.data && e.data.source === iframeSource) {
```

Comment:
I'm assuming that validating e.data.source === iframeSource may implicitly assume that e.data.data exists and that it's a string, but it still leaves room for unreliability.

It's just safer to avoid any assumption and to explicitly validate the specific parameters or variables that are being used within the logical block.

My suggestion is to explicitly validate that e.data.data exists and that it in fact is a string, by making explicit validations, the code is future proof, even if for some reason APIs change down the road.

```suggestion
      if (e?.data && e.data.source === iframeSource && typeof e.data?.data === 'string') {
```